Dockerfile

```diff
@@ -1,26 +1,27 @@
-FROM apache/airflow:1.10.13-python3.8
+FROM apache/airflow:slim-2.7.2-python3.10
 USER root
-ENV POETRY_CACHE_DIR='/var/cache/pypoetry' \
+ENV POETRY_CACHE_DIR='~/.cache/pypoetry' \
     GOOGLE_APPLICATION_CREDENTIALS='/usr/local/airflow/service-account.json'
 
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 467B942D3A79BD29 \
-    && apt-get update \
-    && apt-get install -y --no-install-recommends git \
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends git gcc \
     # 1. if you don't need postgres, remember to remove postgresql-dev and sqlalchemy
     # 2. libglib2.0-0 libsm6 libxext6 libxrender-dev libgl1-mesa-dev are required by opencv
     # 3. git is required by pip install git+https
-    && pip install --no-cache-dir -U poetry==1.6.1 \
+
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
+USER airflow
 COPY pyproject.toml pyproject.toml
 COPY poetry.lock poetry.lock
 
-RUN python -m poetry install --no-interaction --no-ansi --no-dev \
-    # Cleaning poetry installation's cache for production:
-    && rm -rf "$POETRY_CACHE_DIR" \
-    && pip uninstall -yq poetry
+RUN pip install --no-cache-dir -U pip poetry==1.6.1 \
+    && python -m poetry export -o requirements.txt \
+    && python -m pip uninstall poetry -y \
+    && pip install -r requirements.txt
```

Comment:
Use the official constraints here?

https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

Reply:
hmmm... i think we want to manage it through poetry which does not support constraints

Reply:
another way might be to remove airflow from requirements.txt and install the constraint file separately. not sure whether we want to do it though